node/src/proxy_server/http_protocol_pack.rs

```diff
@@ -182,31 +186,31 @@ mod tests {
 
         let host = HttpProtocolPack {}.find_host(&data).unwrap();
 
-        assert_eq!(String::from("192.168.1.230"), host.name);
-        assert_eq!(None, host.port);
+        assert_eq!(host, Host::new("192.168.1.230", HTTP_PORT));
     }
 
     #[test]
-    fn returns_host_name_and_port_from_url_if_both_exist() {
+    fn returns_host_name_and_port_from_header_if_both_exist() {
         let data = PlainData::new(b"OPTIONS http://top.host.com:1234/index.html HTTP/1.1\r\nHost: header.host.com:5432\r\n\r\nbodybody");
 
         let host = HttpProtocolPack {}.find_host(&data).unwrap();
 
-        assert_eq!(String::from("top.host.com"), host.name);
-        assert_eq!(Some(1234), host.port);
+        assert_eq!(host, Host::new("header.host.com", 5432));
     }
 
     #[test]
     fn returns_host_name_from_http_url_if_header_doesnt_exist() {
+        // Note: that "Host: body.host.com" looks like a header, but it's not: it's content.
         let data = PlainData::new(b"DELETE http://top.host.com/index.html HTTP/1.1\r\nContent-Length: 23\r\n\r\nHost: body.host.com\r\n\r\n");
 
         let host = HttpProtocolPack {}.find_host(&data).unwrap();
 
-        assert_eq!(String::from("top.host.com"), host.name);
+        assert_eq!(host.name, String::from("top.host.com"));
```

Comment:
This is old, old code, back from the days when Rust followed the JUnit convention of `assert_eq!(expected, actual)`. Where I happened across these, I changed them to conform to the more modern `assert_eq!(actual, expected)` standard.